lib/Code/TidyAll/Git/Precommit.pm

```diff
@@ -41,8 +41,8 @@ sub check {
         # Store the stash, and restore it upon exiting this scope
         unless ( $self->no_stash ) {
             run( $self->git_path, "stash", "-q", "--keep-index" );
-            scope_guard { run( $self->git_path, "stash", "pop", "-q" ) };
         }
+        scope_guard { run( $self->git_path, "stash", "pop", "-q" ) unless $self->no_stash };
```

Comment:
Note that this is the `Guard` module not the `Scope::Guard` module. This confused me too (I was reading the docs for the wrong module at first). Here:

https://metacpan.org/pod/Guard

I did test this both in my current implementation and in a separate script to verify that it was working the way I expected it to with lexical scopes.

That said I think that the code you are proposing reads more cleanly without the weird `unless` check inside the block so I will try that as well.